File: src/database/DBDefaultRoute.java
package database;

import model.DefaultRoute;
import model.TrailerType;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;

/**
 * DBDefaultRoute
 * Handles all database functionality for default routes.
 *
 * @author Dani Sander
 * @version 1.0
 * @since 20-05-15
 */

public class DBDefaultRoute {

    private DBConnection dbConnection;
    private static DBDefaultRoute instance;

    /**
     * Private constructor for singleton.
     *
     * @throws SQLException
     * @throws ClassNotFoundException
     */
    private DBDefaultRoute() throws ClassNotFoundException, SQLException {
        dbConnection = DBConnection.getInstance();
    }

    /**
     * Singleton method for class.
     *
     * @return instance of class.
     * @throws SQLException
     * @throws ClassNotFoundException
     */
    public static DBDefaultRoute getInstance() throws ClassNotFoundException, SQLException {
        if (instance == null) {
            instance = new DBDefaultRoute();
        }

        return instance;
    }

    /**
     * @return list of all default routes
     */

	public ArrayList<DefaultRoute> getDefaultRoutes() {
        ArrayList<DefaultRoute> list;
        String sql = "select * from DefaultRoute";
//        list = (ArrayList<DefaultRoute>) dbConnection.sendSQL(this, sql, "_formatDefaultRoute");
        return null;
    }

    /**
     * @param rs takes the ResultSet from database
     * @return list of all default routes
     */
    public ArrayList<DefaultRoute> _formatDefaultRoute(ResultSet rs) {
        ArrayList<DefaultRoute> tableList = new ArrayList<DefaultRoute>();
        try {
            while (rs.next()) {
                tableList.add(new DefaultRoute(
                        rs.getLong("id"),
                        rs.getTime("time_of_departure"),
                        TrailerType.valueOf(rs.getString("trailer_type")),
                        rs.getBoolean("extra_route")
                ));
            }
        } catch (SQLException e) {
            // TODO Auto-generated catch block
            e.printStackTrace();
        }
        return tableList;
    }

    public void storeDefaultRoute(DefaultRoute defaultRoute) {
        String sql = String.format("INSERT into Route values('%s', '%s', %d);",
                defaultRoute.getTrailerType().toString(),
                defaultRoute.getTimeOfDeparture().toString(),
                // inline if statement: if true return 1 else return 0
                defaultRoute.isExtraRoute() ? 1 : 0);
        long defaultRouteID = dbConnection.sendInsertSQL(sql);
        defaultRoute.setId(defaultRouteID);
    }
}

File: Server/Database/DBDefaultRoute.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Model;
using System.Data;

namespace Server.Database
{
    class DBDefaultRoute
    {
        private DBConnection DbConnection { get; private set; }
        private static DBDefaultRoute instance; 

        /// private constructor for singelton     
        private DBDefaultRoute()
        {
            DbConnection = DBConnection.Instance;
        }

        /// singelton get instance method 
        /// returns the instance from DB 
        public static DBDefaultRoute Instance
        {
            get
            {
                if (instance == null)
                {
                    instance = new DBDefaultRoute();
                }

                return instance;
            }

        }



        /// Returns a list of all default routes.
        public List<DefaultRoute> DefaultRoutes()
        {
            List<DefaultRoute> list;
            String sql = "select * from DefaultRoute where extra_route=0";
            list = DbConnection.SendSQL<DefaultRoute>(sql, ConvertToDefaultRoute);
            return list;
        }

        /// Takes the ReulstSet from database and returns a list of all default routes.
        public List<DefaultRoute> ConvertToDefaultRoute(IDataReader dataSet)
        {
            List<DefaultRoute> tableList = new List<DefaultRoute>();
            while (dataSet.Read())
            {
                tableList.Add(new DefaultRoute(
                    dataSet.GetInt64(0),
                    dataSet.GetDouble(1),
                    dataSet.GetBoolean(2)
                ));
            }
            return tableList;
        }
    
        /// <summary>
        /// Stores all defaultRoutes in the database.
        /// </summary>
        /// <param name="defaultRoute"></param>
        public void store(DefaultRoute defaultRoute)
        {
            String sql = String.Format("INSERT into DefaultRoute values('{0}', {1});",
                    defaultRoute.TrailerType,
                    // inline if statement: if true return 1 else return 0
                    defaultRoute.isExtraRoute() ? 1 : 0);
            ulong defaultRouteID = DbConnection.SendInsertSQL(sql);
            defaultRoute.ID = defaultRouteID;
        }
    }
}